Jenkinsfile

```diff
@@ -17,6 +17,37 @@
 // is landed, both PR branches can be deleted.
 //@Library(value="pipeline-lib@my_branch_name") _
 
+job_status_internal = [:]
+
+def job_status_write() {
+    if (!env.DAOS_STACK_JOB_STATUS_DIR) {
+        return
+    }
+    String job_name = env.JOB_NAME.replace('/', '_')
+    job_name += '_' + env.BUILD_NUMBER
+    String file_name = env.DAOS_STACK_JOB_STATUS_DIR + '/' + job_name
+
+    String job_status_text = writeYaml data: job_status_internal,
+                                       returnText: true
+
+    // Need to use shell script for creating files that are not
+    // in the workspace.
+    sh label: "Write jenkins_job_status ${file_name}",
+       script: "echo \"${job_status_text}\" >> ${file_name}"
+}
+
+def job_status_update(String name=env.STAGE_NAME,
+                      value=currentBuild.currentResult) {
+    name = name.replace(' ', '_')
+    name = name.replace('.', '_')
+    job_status_internal[name] = value
+}
+
+def job_step_update(value) {
+    // Wrapper around a pipeline step to obtain a status.
+    job_status_update(env.STAGE_NAME, value)
+}
+
```

Comment:
How does all of this get into other projects?

Reply:
It needs to be in the Jenkinsfile, that is the only place simple global "variables" now work.

Reply:
:frowning: